Source/WebCore/Modules/indexeddb/server/MemoryCursor.h

```diff
@@ -35,19 +36,22 @@ class IDBResourceIdentifier;
 
 namespace IDBServer {
 
-class MemoryCursor {
+class MemoryBackingStoreTransaction;
+
+class MemoryCursor : public RefCountedAndCanMakeWeakPtr<MemoryCursor> {
     WTF_MAKE_TZONE_ALLOCATED(MemoryCursor);
 public:
     virtual ~MemoryCursor();
 
     virtual void currentData(IDBGetResult&) = 0;
     virtual void iterate(const IDBKeyData&, const IDBKeyData& primaryKey, uint32_t count, IDBGetResult&) = 0;
 
-    static MemoryCursor* cursorForIdentifier(const IDBResourceIdentifier&);
+    IDBCursorInfo info() const { return m_info; }
 
 protected:
-    MemoryCursor(const IDBCursorInfo&);
+    MemoryCursor(const IDBCursorInfo&, MemoryBackingStoreTransaction&);
 
+private:
```

Comment:
I'm curious--this `m_info` was `protected` before--why do we want to change it to `private`?

Reply:
We prefer not using protected for data members in WebKit:
https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines#do-not-use-protected-visibility-for-data-members